doc/builders/trivial-builders.chapter.md

```diff
@@ -67,99 +101,178 @@ writeTextFile {
-`concatText` and`concatScript` are simple wrappers over `concatTextFile`.
+There are a range of functions that let you concatenate a series of files into a single file within the Nix store, starting with `concatTextFile`.
 
-Here are a few examples:
-```nix
+It accepts one argument as an attribute set containing the following names:
+
+* `name` is the name that Nix will append to the store path in the same way that `stdenv.mkDerivation` uses its `name` attribute.
+* `files` is a list of file paths to be concatenated.
+* `executable` is an optional boolean flag indicating if the output file should be executable.
+  It is `false` by default.
+* `destination` is an optional relative path to be appended to the store path when adding the file to the Nix store.
+* `checkPhase` is an optional script that can be run to verify the output file.
+  It is empty by default and is typically used to perform syntax checks.
+* `meta` is an attribute set which is passed through and corresponds to the `meta` argument for `stdenv.mkDerivation`.
 
+Here are some examples:
+
+```nix
 # Writes my-file to /nix/store/<store path>
+
 concatTextFile {
   name = "my-file";
   files = [ drv1 "${drv2}/path/to/file" ];
 }
-# See also the `concatText` helper function below.
 
 # Writes executable my-file to /nix/store/<store path>/bin/my-file
+
 concatTextFile {
   name = "my-file";
   files = [ drv1 "${drv2}/path/to/file" ];
   executable = true;
   destination = "/bin/my-file";
 }
+```
+
+Building on `concatTextFile` there are a series of simplifications which tailor it to a particular use case and are listed below.
+
+### `concatText` {#trivial-builder-concatText}
+
+```nix
 # Writes contents of files to /nix/store/<store path>
+
 concatText "my-file" [ file1 file2 ]
+```
+
+### `concatScript` {#trivial-builder-concatScript}
+
+```nix
+# Writes contents of files to /nix/store/<store path> and makes it executable
 
-# Writes contents of files to /nix/store/<store path>
 concatScript "my-file" [ file1 file2 ]
 ```
 
 ## `writeShellApplication` {#trivial-builder-writeShellApplication}
 
-This can be used to easily produce a shell script that has some dependencies (`runtimeInputs`). It automatically sets the `PATH` of the script to contain all of the listed inputs, sets some sanity shellopts (`errexit`, `nounset`, `pipefail`), and checks the resulting script with [`shellcheck`](https://github.com/koalaman/shellcheck).
+This function accepts one argument as an attribute set containing the following names:
 
-For example, look at the following code:
+* `name` is the name that Nix will append to the store path in the same way that `stdenv.mkDerivation` uses its `name` attribute.
+* `text` is the contents of the script.
+* `runtimeInputs` is a list of derivations.
+  For each runtime input `/nix/store/<store path>/bin` will be added to the `PATH` of the script before executing the contents of `text`.
```

Comment:
```suggestion
* `runtimeInputs` is a list of derivations.
  For each runtime input `getBin` of it will be prepended to the `PATH` at the start of the script.
```